lib/api/v3/principals/principal_representer_factory.rb

```diff
@@ -35,23 +35,76 @@ class PrincipalRepresenterFactory
         ##
         # Create the appropriate subclass representer
         # for each principal entity
-        def self.create(model, *args)
+        def self.create(model, **args)
           representer_class(model)
-            .create(model, *args)
+            .create(model, **args)
         end
 
         def self.representer_class(model)
           case model.type
-          when 'User'
+          when User.name, DeletedUser.name
```

Comment:
I think now it makes sense to just use this

```
case model
when User
when Group
when PlaceholderUser
```

because then DeletedUser will match User

Reply:
I changed the code accordingly. Although technically, this is not equal as it will now allow the other subclasses of User (e.g. SystemUser) to be matched. But that probably is just a theoretical problem.